pkg/controller/common/helpers.go

```diff
@@ -1001,3 +1008,144 @@ func GetDefaultBaseImageContainer(cconfigspec *mcfgv1.ControllerConfigSpec) stri
 	}
 	return cconfigspec.OSImageURL
 }
+
+// ProxyValidation inspects the proxyTestImage to validate if the http proxy is valid.
+func ProxyValidation(proxy *apicfgv1.ProxyStatus, releaseVerision string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) error {
+	if proxy == nil {
+		return nil
+	}
+
+	oc, err := exec.LookPath("oc")
+	if err != nil {
+		return err
+	}
+
+	var icspfile string
+	if len(icspRules) > 0 {
+		icspfile, err = generateICSPFile(icspRules)
+		if err != nil {
+			return err
+		}
+		// cmd = fmt.Sprintf(imageInfoWithICSP, releaseVersion, tagName, icspfile)
+		defer os.Remove(icspfile)
+	}
+
+	cnoImg, err := cnoImageSpec(oc, releaseVerision, icspfile)
+	if err != nil || cnoImg == "" {
+		return fmt.Errorf("%w: error retrieving CNO image pull spec", err)
+	}
+
+	return ocImageInfoUseProxy(oc, cnoImg, proxy.HTTPProxy, proxy.HTTPSProxy, icspfile)
+}
+
+func cnoImageSpec(oc, releaseVersion, icspfile string) (string, error) {
+	const (
+		tagName           = "cluster-network-operator"
+		imageInfo         = "adm release info %s --image-for %s"
+		imageInfoWithICSP = "adm release info %s --image-for %s --icsp-file %s"
```

Comment:
Could you explain how the ICSP affects the imagespec here?

Reply:
the oc command can accept the icsp file as an alternative source to retrieve the release image. https://github.com/openshift/oc/blob/3cdf3c29f0c109c94eb67124548a6b21fc5f6a22/pkg/cli/admin/release/info.go#L136. 
If the icsp has been configured on the cluster I think we should pass them to the oc to get the release image and get the cno pull spec.